docs/latest/design/extending-envoy-gateway.md

```diff
@@ -0,0 +1,147 @@
+
+As outlined in the [official goals][] for the Envoy Gateway project, one of the main goals is to "provide a common foundation for vendors to build value-added products 
+without having to re-engineer fundamental interactions". Development of the Envoy Gateway project has been focused on developing the core features for the project and 
+K8s Gateway API conformance. This system focuses on the “common foundation for vendors” component by introducing a way for vendors to extend Envoy Gateway.
+
+## Goals
+
+- Provide a foundation for extending the Envoy Gateway control plane
+- Allow users to attach [extensionRef HTTPRoute filters][] to resources introduced and handled by Envoy Gateway extensions.
+- Extension developers should **NOT** have to maintain a custom fork of Envoy Gateway
+- Extension developers should **NOT** require expert knowledge of all of Envoy Gateway's 
+internal components to meaningfully extend the control plane.
+- Provide a system for extending Envoy Gateway which allows extension projects to ship updates independent of Envoy Gateway's release schedule
+- Extensions can hook into both the Envoy configuration pipeline and the infra manager pipeline
+- Setup a foundation for the initial iteration of Extending Envoy Gateway
+
+## Non-Goals
+
+- Address every edge case for extending Envoy Gateway in the initial iteration
+- Extend [Gateway API Policy Attachments][]. At some point these will be addressed, but the initial implementation of the design omits these.
+
+## Overview
+
+TODO: I feel like this section could be better...
+
+Envoy Gateway provides a series of pre/post hooks before and after it's major components (translator, etc.) to allow extensions to modify the data going into or coming out of these components.
+An extension can be created external to Envoy Gateway as it's own deployment or loaded as a sidecar and will communicate over gRPC to receive data as well as context information and then reply 
+with modified data to be used by Envoy Gateway. As extensions fundamentally alter the logic and data that Envoy Gateway provides, Extension projects assume responsibility for any bugs and issues 
+they create as a direct result of their modification of Envoy Gateway.
+
+## Registering Extensions in Envoy Gateway
+
+Information about the extensions that Envoy Gateway needs to load is configured in the Envoy Gateway config. Each extension must specify:
+
+- The name to register the extension under
+- Which API groups the extension supports
+- Extension service connection info (host, post, TLS settings, etc.)
+
+An example configuration:
+
+```yaml
+apiVersion: config.gateway.envoyproxy.io/v1alpha1
+kind: EnvoyGateway
+extensions:
+  - name: dummy-extension
+    apiGroups:
+      - example.myextension.io
+    service:
+      host: dummy-extension.example
+      port: 443
+      tls:
+        type: secret
+        secret:
+            name: dummy-extension
+            namespace: example
+```
+
```

Comment:
@alexgervais - This is what I was talking about, your suggestion is to leave this type of discussion off from the design doc and instead inject it when there are comments and questions?

Reply:
Yes. The design doc should be an implementation guide with facts and straight-up capture decisions and end-state (just like executable code). Discussions, questions and considerations are better handled through GitHub comments and feedback mechanisms. I have no issues with us opening/leading/injecting the discussion and requesting feedback tho!

Reply:
I'll add these points to the PR body instead of the doc since they are important to surface, but these are details that are probably more relevant to the other maintainers between now and the point that the support for this is merged rather than something that is super relevant to the end-user.